From 5ab0b43ce67d8d131242542a80b6aa6b129ed5ee Mon Sep 17 00:00:00 2001
From: Niels de Vos <ndevos@redhat.com>
Date: Mon, 17 Oct 2016 11:58:30 +0200
Subject: [PATCH] gfapi: add glfs_free() to glfs.h

Commit 4721188a154acd9a0a4c096d8d73e97f3bf1b2a9 introduces glfs_free()
but missed adding the function to the header. The symbol is correctly
available in the library though. Testcases do not seem to fail when a
function is missing for the headers...

The glusterfs-3.7.16 packages have been released with the missing
declaration in the header and symbol-maps. Still, the function is
available for applications:

    $ objdump -T usr/lib64/libgfapi.so.0 | grep -w glfs_free
    0000000000006aa0 g    DF .text	0000000000000035  GFAPI_3.7.16 glfs_free

Change-Id: Ia707ee957f090dbfca028192fcc81a83dfdf4ae0
BUG: 1344714
Reported-by: Jiffin Tony Thottan <jthottan@redhat.com>
Signed-off-by: Niels de Vos <ndevos@redhat.com>
Reviewed-on: http://review.gluster.org/15653
CentOS-regression: Gluster Build System <jenkins@build.gluster.org>
Smoke: Gluster Build System <jenkins@build.gluster.org>
Reviewed-by: jiffin tony Thottan <jthottan@redhat.com>
NetBSD-regression: NetBSD Build System <jenkins@build.gluster.org>
---

diff --git a/api/src/gfapi.aliases b/api/src/gfapi.aliases
index 74ef4d3..d2c4ba2 100644
--- a/api/src/gfapi.aliases
+++ b/api/src/gfapi.aliases
@@ -141,6 +141,7 @@
 
 _pub_glfs_h_lookupat _glfs_h_lookupat$GFAPI_3.7.4
 
+_pub_glfs_free _glfs_free$GFAPI_3.7.16
 _pub_glfs_h_poll_upcall _glfs_h_poll_upcall$GFAPI_3.7.16
 _pub_glfs_upcall_get_fs _glfs_upcall_get_fs$GFAPI_3.7.16
 _pub_glfs_upcall_get_reason _glfs_upcall_get_reason$GFAPI_3.7.16
diff --git a/api/src/gfapi.map b/api/src/gfapi.map
index 964db76..d022380 100644
--- a/api/src/gfapi.map
+++ b/api/src/gfapi.map
@@ -175,6 +175,7 @@
 
 GFAPI_3.7.16 {
 	global:
+		glfs_free;
 		glfs_upcall_get_fs;
 		glfs_upcall_get_reason;
 		glfs_upcall_inode_get_event;
diff --git a/api/src/glfs.h b/api/src/glfs.h
index c81b7cc..903492a 100644
--- a/api/src/glfs.h
+++ b/api/src/glfs.h
@@ -769,6 +769,9 @@
 glfs_fd_t *glfs_dup (glfs_fd_t *fd) __THROW
         GFAPI_PUBLIC(glfs_dup, 3.4.0);
 
+void glfs_free (void *ptr) __THROW
+        GFAPI_PUBLIC(glfs_free, 3.7.16);
+
 /*
  * Nobody needs this call at all yet except for the test script.
  */
